vmdb/app/models/mixins/naming_sequence_mixin.rb

```diff
@@ -0,0 +1,19 @@
+module NamingSequenceMixin
+  extend ActiveSupport::Concern
+
+  NAMING = 'naming'
+
+  included do
+    # these are possible naming sequences because there is no constraint to source column
+    has_many :naming_sequences, :as => :resource, :dependent => :destroy, :class_name => "CustomAttribute"
+  end
+
+  def next_naming_sequence(name, source)
+    lock do
+      record = CustomAttribute.where(:name => name, :source => source).first_or_create
```

Comment:
You may still want to access it via naming_sequences.  For one, then it's attached to the region...the code as is won't attach to the region, it will just float, and the has_many will not work.

```ruby
record = naming_sequences.where(:name => name, :source => source).first_or_create
```

Additionally, since you are going to update the attributes anyway, no need to do a create then save...it's a double hit on the db.  Instead, maybe `first_or_initialize`?  I know that `.first_or_initialize` doesn't add it to the association which is kind of annoying.  And for some reason there's no `first_or_build` method (probably because we're on a AR:Relation chain which by now has lost the association?).  Not sure what the right approach is with respect to this, but admittedly it's minor.  cc @jrafanie @matthewd @tenderlove